core/src/main/java/org/elasticsearch/index/engine/LiveVersionMap.java

```diff
@@ -117,7 +117,7 @@ public void afterRefresh(boolean didRefresh) throws IOException {
         // case.  This is because we assign new maps (in beforeRefresh) slightly before Lucene actually flushes any segments for the
         // reopen, and so any concurrent indexing requests can still sneak in a few additions to that current map that are in fact reflected
         // in the previous reader.   We don't touch tombstones here: they expire on their own index.gc_deletes timeframe:
-        maps = new Maps(maps.current, ConcurrentCollections.<BytesRef,VersionValue>newConcurrentMapWithAggressiveConcurrency());
+        maps = new Maps(maps.current, Collections.emptyMap());
```

Comment:
I'm probably lacking the larger context here but is it ok to change the implementation class from `ConcurrentHashMap` to `HashMap` here?

Reply:
yeah it's an immutable map and we should never ever modify it too.